### PR TITLE
Fix inability for credbot to push to master in Release workflow

### PR DESCRIPTION
**Pull request by hammadj**
Updated the workflow to fetch all history for commits which is required for the refs to properly be pushed into
 master according to the docs of the push action (https://github.com/ad-m/github-push-action). Also disables
 persist-credentials which prevents the auto-generated GITHUB_TOKEN for actions from being used.

Verified successful run on a fork with same repo security settings (https://github.com/hammadj/sourcecred/runs/1098839653?check_suite_focus=true)

Test Plan: Create a release and ensure the action runs successfully and updates master branch with new version.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -20,14 +23,14 @@ jobs:
       - name: Test
         run: yarn test --full --ci
         env:
-          SOURCECRED_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SOURCECRED_GITHUB_TOKEN: ${{ secrets.SOURCECRED_GITHUB_TOKEN }}
```
Comment (decentralion) [inline .github/workflows/publish.yml:26]: Why this change? If we want the Credbot token for its permissions, we could use `CREDBOT_TOKEN`. But since this is for `yarn test`, the default GITHUB_TOKEN should be fine, no?

**Review by decentralion — COMMENTED** [review]
After this change, do we need the CREDBOT admin token to be present at all?

File: .github/workflows/publish.yml (hunk 1 above)
Comment (hammadj) [reply .github/workflows/publish.yml:26]: Adding `persist-credentials: false` disables the default GITHUB_TOKEN that goes into Actions, so changed it to use this one instead.

**Comment by hammadj** [comment]
> After this change, do we need the CREDBOT admin token to be present at all?

Yea we still need it there

File: .github/workflows/publish.yml (hunk 1 above)
Comment (decentralion) [reply .github/workflows/publish.yml:26]: Got it.

**Review by decentralion — APPROVED** [review]
Will cut a release today most likely, so we'll get to test this in prod soon. :)
